solidity-frontend/src/main/java/org/sonarsource/solidity/frontend/Utils.java

```diff
@@ -97,4 +102,41 @@ public static boolean isCommentSignificant(String token) {
     return matcher.find();
   }
 
+  public static boolean isTernaryExpression(StatementContext ctx) {
+    // TODO
+    String statementName = ctx.getChild(0).getClass().getSimpleName();
+    ExpressionContext expr = null;
+    switch (statementName) {
+      case "ReturnStatementContext":
+        ReturnStatementContext retStmt = ctx.returnStatement();
+        expr = retStmt.expression();
+        return (expr.getToken(SolidityParser.TERNARY_OPERATOR, 0) != null);
+      case "SimpleStatementContext":
+        SimpleStatementContext simpleStmt = ctx.simpleStatement();
+        VariableDeclarationStatementContext varDeclStmt = simpleStmt.variableDeclarationStatement();
+        if (varDeclStmt != null) {
+          expr = varDeclStmt.expression();
+          return (expr.getToken(SolidityParser.TERNARY_OPERATOR, 0) != null);
+        }
+      default:
+    }
+    return false;
+  }
+
+  public static ExpressionContext countTernaryExpressionOperators(StatementContext ctx) {
+    String statementName = ctx.getChild(0).getClass().getSimpleName();
+    switch (statementName) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this "switch" statement by "if" statements to increase readability. ([squid:S1301](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1301))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=22&issues=AWQ2SgNkYwZU9vvru9od&open=AWQ2SgNkYwZU9vvru9od)

[ISSUE_KEY]: AWQ2SgNkYwZU9vvru9od